### PR TITLE
 Fix for: Reset Bookmark Feature Throws Error #57

### PR DESCRIPTION
**Pull request by iamrahulrnair**
Hi, I have replaced the BrowserRouter to **HashRouter**, 404 page from github was due to the server not being not able to understand  the subdomain with /bookmarks, only /gocanalyzer is configured in github pages, setting hashrouter remove the post # characters and possibly sending the request to github pages server as https://sparsh1212.github.io/gsocanalyzer/ hence the index.html file is obtained.😊

### CODE REVIEW

File: src/components/OrganisationCard.js [hunk 1]
```diff
@@ -45,12 +45,12 @@ const OrganisationCard = ({key, orgData, bookmarked, setBookmarked}) => {
                 }
               }
               setBookmarked([...array]);
-            }} title="Delete Bookmark"/>
+            }} title="Delete Bookmark" alt=""/>
```
Comment (Sparsh1212) [inline src/components/OrganisationCard.js:48]: Why is this added? 
Ditto for others too.

File: src/App.js [hunk 2]
```diff
@@ -15,6 +15,7 @@ function App() {
   }, [])
 
   useEffect(() => {
+    console.log("Deployed")
```
Comment (Sparsh1212) [inline src/App.js:18]: Please remove log statements.

File: src/components/AdvancedSearch.js [hunk 3]
```diff
@@ -115,7 +115,7 @@ const AdvancedSearch = forwardRef((props, ref) => {
 
   useEffect(() => {
     autoComplete && setSuggestions(autoComplete.suggest(search))
-  }, [search])
+  }, [search,autoComplete])
```
Comment (Sparsh1212) [inline src/components/AdvancedSearch.js:118]: Can you explain, why this is needed?

File: src/components/Footer.js [hunk 4]
```diff
@@ -9,7 +9,7 @@ const Footer = () => {
             <h3>
                 Made with <FontAwesomeIcon color='red' icon={faHeart} /> by Sparsh
                 Agrawal and Rishabh Dugaye |{'  '}
-                <a href='https://github.com/Sparsh1212' target='_blank'>
+                <a href='https://github.com/Sparsh1212' target='_blank' rel="noopener noreferrer">
```
Comment (Sparsh1212) [inline src/components/Footer.js:12]: Why does this need to be added?

File: src/components/OrganisationCard.js (hunk 1 above)
Comment (iamrahulrnair) [reply src/components/OrganisationCard.js:48]: Its shows warning, warning can be removed by adding an empty alt="" in img tags

**Review by Sparsh1212 — CHANGES_REQUESTED** [review]
@iamrahulrnair Please take a look at the comments added and also can you please attach a demo video that proves that it is fixing the bug.

File: src/App.js (hunk 2 above)
Comment (iamrahulrnair) [reply src/App.js:18]: will do that😉

File: src/components/AdvancedSearch.js (hunk 3 above)
Comment (iamrahulrnair) [reply src/components/AdvancedSearch.js:118]: dependency warning is removed, checked autocomplete value don't change often, so I think it don't have any after effects

File: src/components/OrganisationCard.js (hunk 1 above)
Comment (Sparsh1212) [reply src/components/OrganisationCard.js:48]: In that case, it will be better to give something relevant like:
`alt="Delete Bookmark"`
Ditto for all alt tags added.

File: src/components/Footer.js (hunk 4 above)
Comment (iamrahulrnair) [reply src/components/Footer.js:12]: this is a vulnerability warning the linter throws, you can read more about [here](https://www.jitbit.com/alexblog/256-targetblank---the-most-underestimated-vulnerability-ever/).

**Comment by Sparsh1212** [comment]
Also please revert all the linter warning fixes. Let's keep this PR to what the issue is about. Once, this gets fixed, we can file a separate issue to take care of all the warnings.

**Comment by iamrahulrnair** [comment]
> Also please revert all the linter warning fixes. Let's keep this PR to what the issue is about. Once, this gets fixed, we can file a separate issue to take care of all the warnings.

fix done 😄

**Comment by iamrahulrnair** [comment]
> @iamrahulrnair Please take a look at the comments added and also can you please attach a demo video that proves that it is fixing the bug.

This fix can only be tested on deployment, this is specific to GitHub-pages.

File: src/components/Footer.js [hunk 5]
```diff
@@ -35,4 +35,4 @@ const Footer = () => {
     );
 }
 
-export default Footer;
\ No newline at end of file
+export default Footer;
```
Comment (Sparsh1212) [inline src/components/Footer.js]: Redundant change. Please remove.

**Review by Sparsh1212 — CHANGES_REQUESTED** [review]
Please revert changes to `package-lock.json`. Those are automatically done by ide.

**Comment by Sparsh1212** [comment]
> Hi, I have replaced the BrowserRouter to **HashRouter**, 404 page from github was due to the server not being not able to understand the subdomain with /bookmarks, only /gocanalyzer is configured in github pages, setting hashrouter remove the post # characters and possibly sending the request to github pages server as https://sparsh1212.github.io/gsocanalyzer/ hence the index.html file is obtained.😊

Out of curiosity, can please share any reference from where you got that insight. I'd love to read more about it.

**Comment by iamrahulrnair** [comment]
> > Hi, I have replaced the BrowserRouter to **HashRouter**, 404 page from github was due to the server not being not able to understand the subdomain with /bookmarks, only /gocanalyzer is configured in github pages, setting hashrouter remove the post # characters and possibly sending the request to github pages server as https://sparsh1212.github.io/gsocanalyzer/ hence the index.html file is obtained.😊
> 
> Out of curiosity, can you please share any reference from where you got that insight? I'd love to read more about it.

check [this](https://create-react-app.dev/docs/deployment/#github-pages), read notes on client-side-routing

File: src/components/Footer.js (hunk 5 above)
Comment (iamrahulrnair) [reply src/components/Footer.js]: I cant figure out how to resolve this, any idea?

Comment (Sparsh1212) [inline src/components/Footer.js]: This is typically the case when you insert an extra line at the end of the file. The line ending is an invisible character and because of that, you are not able to see it in git diff.
So, either you have removed or added a line ending. I am guessing, you must have added one. So please try reverting that and let me know if it works.

Comment (iamrahulrnair) [reply src/components/Footer.js]: its not working i've tried removing the content and adding the whole content, tried removing the last line also, not working,
can you revert those change when merging?

Comment (Sparsh1212) [reply src/components/Footer.js]: That's strange. Alright, we can keep it in the PR.

**Review by Sparsh1212 — APPROVED** [review]

**Comment by Sparsh1212** [comment]
LGTM!
